heroes-app/src/app/add-hero/add-hero.component.ts

```diff
@@ -0,0 +1,34 @@
+import { Component, OnInit } from '@angular/core';
+import { Hero } from '../hero';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+import { HeroService } from '../hero.service';
+
+@Component({
+  selector: 'app-add-hero',
+  templateUrl: './add-hero.component.html',
+  styleUrls: ['./add-hero.component.less']
+})
+export class AddHeroComponent implements OnInit {
+
+  heroForm: FormGroup;
+  newHero: Hero;
+  powers = ['Really Smart', 'Super Flexible',
+    'Super Hot', 'Weather Changer'];
+
+  constructor(private heroService: HeroService, private formBuilder: FormBuilder) {
+    this.heroForm = formBuilder.group({
+      name: ['', [Validators.required]],
+      description: ['', [ Validators.required]],
```

Comment:
```suggestion
      description: [null, [ Validators.required]],
```
It's better to use null as default value, because empty string may be a valid value